runtime-modules/content/src/lib.rs

```diff
@@ -449,6 +501,69 @@ decl_module! {
             Self::deposit_event(RawEvent::ChannelUpdated(actor, channel_id, channel, params));
         }
 
+        // Extrinsic for updating channel privilege level (requires lead access)
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_channel_privilege_level(
+            origin,
+            channel_id: T::ChannelId,
+            new_privilege_level: T::ChannelPrivilegeLevel,
+        ) {
+            let sender = ensure_signed(origin)?;
+
+            ensure_lead_auth_success::<T>(&sender)?;
+
+            // check that channel exists
+            Self::ensure_channel_exists(&channel_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Update the channel
+            ChannelById::<T>::mutate(channel_id, |channel| { channel.privilege_level = new_privilege_level });
+
+            Self::deposit_event(RawEvent::ChannelPrivilegeLevelUpdated(channel_id, new_privilege_level));
+        }
+
+        // extrinsics for pausing/re-enabling specified channel features
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn change_channel_features_status_as_moderator(
+            origin,
+            actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+            channel_id: T::ChannelId,
+            channel_feature_status_changes: ChannelFeatureStatusChanges,
+            rationale: Vec<u8>,
+        ) -> DispatchResult {
+
+            let sender = ensure_signed(origin)?;
+            // check that channel exists
+            let channel = Self::ensure_channel_exists(&channel_id)?;
+            let mut new_paused_features = channel.paused_features.clone();
+
+            // Check permissions for moderation actions
+            let mut actions = Vec::<ContentModerationAction>::new();
```

Comment:
If you follow my recommendation on updating this state, I believe most of this goes away.